src/core.js

```diff
@@ -2030,6 +2076,19 @@ define([
     };
 
     /**
+     * Extension point for mug setup during control node parsing
+     *
+     * The mug has been inserted into the tree by the time this method
+     * is called.
```

Comment:
Establishing this invariant makes handling badly written XForms much easier: default itext IDs, which depend on the mug's absolute path, can now be calculated reliably.